libtcmu.c

```diff
@@ -544,34 +544,37 @@ static void remove_device(struct tcmulib_context *ctx,
 static int is_uio(const struct dirent *dirent)
 {
 	int fd;
-	char tmp_path[64];
-	char buf[256];
-	ssize_t ret;
+	char *tmp_path;
+	char buf[256] = {'\0'};
+	ssize_t ret = 0;
 
 	if (strncmp(dirent->d_name, "uio", 3))
 		return 0;
 
-	snprintf(tmp_path, sizeof(tmp_path), "/sys/class/uio/%s/name", dirent->d_name);
+	if (asprintf(&tmp_path, "/sys/class/uio/%s/name", dirent->d_name) == -1)
+		return 0;
 
 	fd = open(tmp_path, O_RDONLY);
 	if (fd == -1)
-		return 0;
+		goto out;
 
 	ret = read(fd, buf, sizeof(buf));
 	if (ret <= 0 || ret >= sizeof(buf))
-		goto not_uio;
+		goto out;
 
 	buf[ret-1] = '\0'; /* null-terminate and chop off the \n */
 
 	/* we only want uio devices whose name is a format we expect */
 	if (strncmp(buf, "tcm-user", 8))
-		goto not_uio;
+		goto out;
 
-	close(fd);
-	return 1;
-not_uio:
-	close(fd);
-	return 0;
+	ret = 1;
+
+out:
+	free(tmp_path);
+	if (fd != -1)
+		close(fd);
+	return ret;
```

Comment:
Make this fit the more common style used. It would look something like:

```
close:
       close(fd);
free_path:
        free(tmp_path);
       return ret;
```